sdk/servicebus/service-bus/src/serviceBusMessage.ts

```diff
@@ -983,42 +935,56 @@ export class ServiceBusMessage implements ReceivedMessage {
    *
    * @returns Promise<void>
    */
-  async deadLetter(options?: DeadLetterOptions): Promise<void> {
-    const error: AmqpError = {
-      condition: Constants.deadLetterName
-    };
-    if (options) {
-      error.info = {
-        DeadLetterReason: options.deadletterReason,
-        DeadLetterErrorDescription: options.deadLetterErrorDescription
-      };
-    }
+  async deadLetter(propertiesToModify?: DeadLetterOptions & { [key: string]: any }): Promise<void> {
     log.message(
       "[%s] Deadlettering the message with id '%s'.",
       this._context.namespace.connectionId,
       this.messageId
     );
-    if (this._context.requestResponseLockedMessages.has(this.lockToken!)) {
-      await this._context.managementClient!.updateDispositionStatus(
-        this.lockToken!,
-        DispositionStatus.suspended,
-        {
-          deadLetterReason: error.condition,
-          deadLetterDescription: error.description,
-          sessionId: this.sessionId
-        }
-      );
 
-      // Remove the message from the internal map of deferred messages
-      this._context.requestResponseLockedMessages.delete(this.lockToken!);
+    const actualPropertiesToModify: Partial<DeadLetterOptions> = {
+      ...propertiesToModify
+    };
+
+    // these two fields are handled specially and don't need to be in here.
+    delete actualPropertiesToModify.deadLetterErrorDescription;
+    delete actualPropertiesToModify.deadletterReason;
+
+    const dispositionStatusOptions: DispositionStatusOptions = {
+      propertiesToModify: actualPropertiesToModify,
+      deadLetterReason: propertiesToModify?.deadletterReason,
+      deadLetterDescription: propertiesToModify?.deadLetterErrorDescription
+    };
+    return this.settleMessage(DispositionStatus.suspended, dispositionStatusOptions);
+  }
+
+  private async settleMessage(
+    operation: DispositionStatus,
+    options?: DispositionStatusOptions
+  ): Promise<void> {
+    const dispositionType = getDispositionType(operation);
+    const isDeferredMessage = this._context.requestResponseLockedMessages.has(this.lockToken!);
+    const receiver = isDeferredMessage
+      ? undefined
+      : this._context.getReceiver(this.delivery.link.name, this.sessionId);
+    if (!isDeferredMessage) this.throwIfMessageCannotBeSettled(receiver, dispositionType!);
+
+    // Message Settlement with managementLink
+    // 1. If the received message is deferred
+    // 2. If the received message is without a receiveLink that needs backup managementLink for message settlement
```

Comment:
Comments are helpful if they provide info that is more than what can be inferred from reading the code itself

```suggestion
    // Message Settlement with managementLink
    // 1. If the received message is deferred as such messages can only be settled using managementLink
    // 2. If the associated receiver link is not available. This does not apply to messages from sessions as we need a lock on the session to do so.
```

Reply:
Thanks.